src/DynamoDbQueryBuilder.php

```diff
@@ -78,6 +79,81 @@ public function take($value)
     }
 
     /**
+     * Alias to set the "offset" value of the query.
+     *
+     * @param  mixed  $value
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function skip($value)
+    {
+        return $this->offset($value);
+    }
+
+    /**
+     * Alias to set the "setLastEvaluatedKey" value of the query.
+     *
+     * @param  mixed  $value
+     * @return $this
+     */
+    public function offset($value)
+    {
+        return $this->setLastEvaluatedKey($value);
+    }
+
+    /**
+     * Set the "lastEvaluatedKey" value of the query.
+     *
+     * @param mixed $value
+     * @return $this
+     */
+    public function setLastEvaluatedKey($value)
+    {
+        if (empty($value)) {
+            $this->lastEvaluatedKey = null;
+            return $this;
+        }
+
+        if ($this->model->hasCompositeKey()) {
+            // require $value to be an array
+            if (!is_array($value) && count($value) < 2) {
+                throw new InvalidArgumentException('$value must be an array with 2 elements when the model uses a Composite Key');
+            }
+
+            $keys = $this->model->getCompositeKeyName();
+            $last = [];
+            $last[array_shift($keys)] = $this->model->getMarshaler()->marshalValue(array_shift($value));
```

Comment:
`array_shift` would mutate `$keys` which is pointing to the model's property `$compositeKey`. I suspect this will have unexpected side effects. I would prefer to not use any functions with side-effects.

Reply:
Can just use a for loop here